onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -5304,6 +5304,17 @@ def aten_swapdims(self: TensorType, dim0: int, dim1: int) -> TensorType:
     raise NotImplementedError()
 
 
+@torch_op("aten::sym_size")
+def aten_sym_size(self: TReal, dim: int = 0) -> TReal:
+    """sym_size(Tensor self, int dim) -> Tensor"""
+    # NOTE: onnx-script doesn't support attribute process,
+    # so op.Shape(self, start=dim, end=dim + 1) is not supported.
+    shape = op.Shape(self)
+    start = op.Reshape(dim, [1])
+    end = op.Reshape(dim + 1, [1])
+    return op.Slice(shape, start, end)
```

Comment:
Shape + Gather?

Reply:
It seems I would have to put dim into a list? 
```python
@torch_op("aten::sym_size")
def aten_sym_size(self: TReal, dim: int = 0) -> TReal:
    """sym_size(Tensor self, int dim) -> Tensor"""
    # NOTE: onnx-script doesn't support attribute process,
    # so op.Shape(self, start=dim, end=dim + 1) is not supported.
    shape = op.Shape(self)
    return op.Gather(shape, [dim])
```

some expression errors

Reply:
a bit more like 
https://github.com/pytorch/pytorch/blob/be220690d9d8b7e8d58b81694c6cbf0958f94845/torch/onnx/symbolic_helper.py#L1322

You can reference https://github.com/pytorch/pytorch/blob/be220690d9d8b7e8d58b81694c6cbf0958f94845/torch/onnx/symbolic_opset9.py#L1019
There are a lot edge case handling which are probably not needed.

Reply:
Revisit on this, and found that in this op, we should follow this line https://github.com/pytorch/pytorch/blob/be220690d9d8b7e8d58b81694c6cbf0958f94845/torch/onnx/symbolic_opset9.py#L1188, but it doesn't seem right.

According to [Gather](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Gather), axis should be int, and `dim` in aten_sym_size should be equivalent to `indices` which is tensor(INT32/INT64). Either way, I need Reshape to convert dim from int to tensor(INT64) I think. So I prefer to keep this function. LMK if I miss anything!

Although I tried like
```python
@torch_op("aten::sym_size")
def aten_sym_size(self: TReal, dim: int = 0) -> TReal:
    """sym_size(Tensor self, int dim) -> Tensor"""
    # NOTE: onnx-script doesn't support attribute process,
    # so op.Shape(self, start=dim, end=dim + 1) is not supported.
    shape = op.Shape(self)
    idx= op.Reshape(dim, [1])
    return op.Slice(shape, idx)
```
It errors with SegFault.

Reply:
> I need Reshape to convert dim from int to tensor(INT64)

Can you use Constant?

> It errors with SegFault.

Does it segfault in onnxscript or onnxruntime? Could you open an issue?

Reply:
> Can you use Constant?

No, I remember @gramalingam told me that `dim` is an expression that we can't prcoess it in attributes of ONNX op. I am not so familiar with the root cause. Maybe @gramalingam or @justinchuby can share more here?

> Does it segfault in onnxscript or onnxruntime? Could you open an issue? 

It seems the same error I mentioned yesterday. The protobuf can't be called on `onnx_model.SerializeToString()`. It's from protobuf API, but I will post it and see if ONNX team would know more about this.

Reply:
> It seems the same error I mentioned yesterday. The protobuf can't be called on onnx_model.SerializeToString(). It's from protobuf API, but I will post it and see if ONNX team would know more about this.

A second thought on this, thinking this might be the pass: `self._torch_graph._export_onnx()` being incompatible with new graph?

Reply:
Yeah Constant requires a compile time constant

Reply:
Ok, there are a couple of different issues here: the SegFault is mysterious and needs to be debugged. The other issue is straightforward: if we want to convert a scalar attribute into a 1D tensor, we need to do it in two steps, first converting it into a 0D tensor (via Constant, but OnnxScript will insert this for you), and then by reshaping it into a 1D tensor.

Reply:
https://github.com/microsoft/onnx-script/issues/523

Reply:
As discussed in another PR, `dim+1` also doesn't work if dim is -1. Hence, Gather seems a better solution.